browser/net/brave_system_request_handler.h

```diff
@@ -18,10 +19,11 @@ namespace brave {
 
 std::string BraveServicesKeyForTesting();
 
-void AddBraveServicesKeyHeader(network::ResourceRequest* url_request);
+void AddBraveServicesKeyHeader(
+    std::shared_ptr<network::ResourceRequest> url_request);
 
-network::ResourceRequest OnBeforeSystemRequest(
-    const network::ResourceRequest& url_request);
+std::shared_ptr<network::ResourceRequest> OnBeforeSystemRequest(
```

Comment:
Not sure if there is any need to use shared pointers. Generally we should avoid them if there is no strong reason.

Reply:
Within `system_request_handler.cc` we operate with incomplete type `network::ResourceRequest`. That's why `shared_ptr` is used.

Reply:
it's not just that we shouldn't use shared pointers, you are taking ownership of something you don't own and this is likely to cause crashes when the shared pointer frees the memory that is actually owned by another class

Reply:
sorry, didn't notice at first that the original ResourceRequest wasn't a pointer so this was creating a copy, but can be fixed by moving to `services/network` as discussed to remove the shared_ptr

Reply:
Moved. Please review.